src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,340 @@
+const MAX_SCRIPT_SIZE = 10000;
+
+/// ScriptBuilder Errors
+const ScriptBuilderError = error{ ScriptTooLong, ScriptTooShort };
+
+///ScriptBuilder is a library to generate easier scripts, useful for faster testing
+pub const ScriptBuilder = struct {
+    /// Dynamic array holding the opcodes
+    script: std.ArrayList(u8),
+
+    ///Memory Allocator
+    allocator: Allocator,
+
+    /// Initialize a new ScriptBuilder
+    ///
+    /// # Arguments
+    /// - `allocator`: Memory allocator for managing engine resources
+    ///
+    /// # Returns
+    /// - `!ScriptBuilder`: use with `try`
+    pub fn new(allocator: Allocator, capacity: usize) !ScriptBuilder {
+        return ScriptBuilder{
+            .script = try std.ArrayList(u8).initCapacity(allocator, capacity),
+            .allocator = allocator,
+        };
+    }
+
+    ///Push an OPcode to the ScriptBuilder
+    pub fn addOpcode(self: *ScriptBuilder, op: Opcode) !*ScriptBuilder {
+        if (self.script.items.len >= MAX_SCRIPT_SIZE) {
+            return ScriptBuilderError.ScriptTooLong;
+        }
+        try self.script.append(op.toBytes());
+        return self;
+    }
+
+    fn i32ToBytes(value: i32) [4]u8 {
+        return [_]u8{
+            @intCast(value & 0xFF), // Lower byte (least significant byte)
+            @intCast((value >> 8) & 0xFF), // Next byte
+            @intCast((value >> 16) & 0xFF), // Next byte
+            @intCast((value >> 24) & 0xFF), // Upper byte (most significant byte)
+        };
+    }
+
+    ///Push an Int to the SciptBuilder
+    pub fn addInt(self: *ScriptBuilder, num: i32) !*ScriptBuilder {
+        var byteArray: [4]u8 = undefined;
+
+        // Get the bytes from the integer and copy them to the array
+        std.mem.copyForwards(u8, &byteArray, std.mem.asBytes(&num));
+
+        // const p = @to(num);
+        if (canonicalDataSize(&byteArray) + self.script.items.len >= MAX_SCRIPT_SIZE) {
+            return ScriptBuilderError.ScriptTooLong;
+        }
+        // return try self.addBytes(std.mem.asBytes(std.mem.nativeToLittle(i32, num)));
+        switch (num) {
+            0 => {
+                try self.script.append(Opcode.OP_0.toBytes());
+            },
+            1...16 => try self.script.append(@intCast(Opcode.OP_1.toBytes() - 1 + num)),
+            else => _ = try self.addData(try ScriptNum.new(num).toBytes(self.allocator)),
+        }
+        return self;
+    }
+
+    // pub fn addBytes(self: *ScriptBuilder, data: []const u8) !*ScriptBuilder {
+    //     try self.script.appendSlice(data);
+    // }
+
+    pub fn addData(self: *ScriptBuilder, data: []const u8) !*ScriptBuilder {
+        const bytes_length: usize = @intCast(canonicalDataSize(data));
+        if (bytes_length + self.script.items.len >= MAX_SCRIPT_SIZE) {
+            return ScriptBuilderError.ScriptTooLong;
+        }
+
+        if (data.len == 0 or data.len == 1 and data[0] == 0) {
+            _ = try self.addOpcode(Opcode.OP_0);
```

Comment:
_ =

Reply:
addOpcode returns !*ScriptBuilder and hence it needs to be loaded as unused variable
Please correct me if I'm wrong